fuse_models/src/unicycle_2d_ignition.cpp

```diff
@@ -227,26 +264,24 @@ void Unicycle2DIgnition::process(const geometry_msgs::PoseWithCovarianceStamped&
   if (!params_.reset_service.empty())
   {
     // Wait for the reset service
-    while (!reset_client_.waitForExistence(rclcpp::Duration::from_seconds(10.0)) && ros::ok())
+    while (!reset_client_->wait_for_service(std::chrono::seconds(10))
+           && interfaces_.get_node_base_interface()->get_context()->is_valid())
     {
-      RCLCPP_WARN_STREAM(node_->get_logger(),
-                         "Waiting for '" << reset_client_.getService() << "' service to become avaiable.");
+      RCLCPP_WARN_STREAM(logger_,
+                         "Waiting for '" << reset_client_->get_service_name() << "' service to become avaiable.");
     }
 
-    auto srv = std_srvs::Empty();
-    if (!reset_client_.call(srv))
-    {
-      // The reset() service failed. Propagate that failure to the caller of this service.
-      throw std::runtime_error("Failed to call the '" + reset_client_.getService() + "' service.");
-    }
+    auto srv = std::make_shared<std_srvs::srv::Empty::Request>();
+    // No need to spin since node is optimizer node, which should be spinning
+    auto result_future = reset_client_->async_send_request(srv);
```

Comment:
Same comment about maybe needing to wait on the result future

Reply:
Fixed! https://github.com/locusrobotics/fuse/pull/304/commits/85aa1bcc392aa0728296864a82d6d01374676c1c